New Feature Flag Tracking page in RUM (#17088)
DataDog/documentation: content/en/real_user_monitoring/feature_flag_tracking/_index.md

- Be confident that your feature will successfully launch without unintentionally causing a bug or performance regression
- Correlate feature releases with performance, pinpoint issues to specific releases, and troubleshoot faster
- Streamline data collection and analysis and focus on troubleshooting

## Set up feature flag data collection

To see detailed set up instructions, see our guide to [get started with feature flag data collection][1]

Feature flag tracking is available in the RUM Browser SDK. To start, set up [RUM browser monitoring][2]. You need the Browser RUM SDK version >= 4.25.0.

You can start collecting feature flag data for [custom feature flag management solutions][3], or using one of our integration partners. 

We currently support integrations with:
- [LaunchDarkly][4]
- [Split][5]
- [Flagsmith][6]

Feature flags will show up in the context of events where they are evaluated, meaning they should show up on the views that the feature flag code logic is run on.

## View your Feature Flags

Once you have set up your feature flag data collection, navigate to the [**Feature Flags**][7] tab within RUM or click the [**Feature Flags**][7] tab under **UX Monitoring**

> From this view you can investigate any questions you have about your feature flag’s health and usage.

Reviewer: ```suggestion
From this view, you can investigate any questions you have about your feature flag’s health and usage.
```

- Monitor the number of users experiencing each variant and see summary statistics of your feature flag
- Check the status of your feature flag to see if there are any that can be removed for code clean up
- View which pages your feature flags are being evaluated on

{{< img src="real_user_monitoring/feature_flag_tracking/feature-flag-list.png" alt="View a list of your feature flags to investigate any questions you have about your feature flag’s health and usage" style="width:90%;" >}}

### Search and filter

> Search and filter your feature flags by typing in the search bar. You can also use the faceted search, to narrow down, broaden, and shift your focus on subsets of feature flags you are interested in.

Reviewer: ```suggestion
Search and filter your feature flags by typing in the search bar. You can also use the faceted search to narrow down, broaden, or shift your focus on subsets of feature flags you are interested in.
```

{{< img src="real_user_monitoring/feature_flag_tracking/feature-flag-list-search-filter.png" alt="Feature Flag list search bar and filtering" style="width:90%;" >}}

### Feature Flag Status
There are two possible feature flag statuses:

> - **Inactive**: For the past 2 weeks there have only been feature flag evaluations for your control variant

Reviewer: ```suggestion
- **Inactive**: For the past 2 weeks, there have only been feature flag evaluations for your control variant
```

> - **Out to 100%**: For the past 2 weeks there have only been feature flag evaluations for one of your non-control variants

Reviewer: ```suggestion
- **Out to 100%**: For the past 2 weeks, there have only been feature flag evaluations for one of your non-control variants
```

## Analyze your feature flags

> To get more details about the health and performance of your feature flag, you can click on the flag in the list to navigate to a dedicated feature flag analysis dashboard. The Feature Flag analysis dashboard provides an overview of the performance of your feature flag, showing you information about user sessions, changes in your Core Web Vitals, and error rates. 

Reviewer: ```suggestion
To get more details about the health and performance of your feature flag, you can click the flag in the list to navigate to a dedicated feature flag analysis dashboard. The Feature Flag analysis dashboard provides an overview of the performance of your feature flag, displaying information about user sessions, changes in your Core Web Vitals, and error rates. 
```

> These out of the box graphs are aggregated across your flag variants, making it easy to spot problems in your feature releases before they turn into serious issues. This dashboard provides an easy way to monitor your feature releases and allows you to quickly roll back as soon as you spot an issue so you can avoid negative user experiences. 

Reviewer: ```suggestion
These out-of-the-box graphs are aggregated across your flag variants, making it easy to spot problems in your feature releases before they turn into serious issues. This dashboard provides an easy way to monitor your feature releases and allows you to quickly roll back as soon as you spot an issue so you can avoid negative user experiences. 
```

The **Issues** tab gives you a view of the errors that are occurring in your application for user sessions that have your feature flag. See if any issues detected by [Error Tracking][7] occurred for a specific variant of your feature flag and may be related to your changes.

The **Performance** tab allows you to understand if one of your feature flag variants have caused poor performance. You can view your Core Web Vitals and loading time for each variant to determine if one of your variants may be causing a negative impact on your application's performance.

### Build custom views from Feature Flag data using the RUM Explorer
Search through all the data collected by RUM in the [RUM Explorer][8] to surface trends on feature flags, analyze patterns with greater context, or export them into [dashboards][9] and [monitors][10]. 

You can search your Sessions, Views, or Errors in the RUM Explorer, with the `@feature_flags.{flag_name}` attribute to scope down and focus on events where users were shown a specific user experience.

> You can compare important metrics to you and your teams by grouping your query by `@feature_flags.{flag_name}`. For example, if we want to understand how our new checkout flow is affecting the conversion rate from our checkout page to users making a purchase, we can add a "Group by" on our conversion rate graph.

Reviewer: ```suggestion
You can compare important metrics to you and your teams by grouping your query by `@feature_flags.{flag_name}`. For example, if you want to understand how your new checkout flow is affecting the conversion rate from the checkout page to users making a purchase, you can add a "Group by" on the conversion rate graph.
```
